client/reader_slice_of_slices_iterator.go

```diff
@@ -98,8 +118,20 @@ func (it *readerSliceOfSlicesIterator) Close() {
 		return
 	}
 	it.closed = true
-	if it.pool != nil {
-		it.pool.Put(it)
+	// Release any refs to segments
+	it.segments = nil
+	// Release any refs to segment byte slices
+	for i := range it.segmentReaders {
+		seg := it.segmentReaders[i].Segment()
+		if seg.Head != nil {
+			seg.Head.Reset(nil)
```

Comment:
So when `Reset` is called we don't actually give up ref to the checked bytes struct itself and we reuse it.  Hence why I never DecRef for the `checked.Bytes` here.